tracetools_trace/tracetools_trace/tools/lttng_impl.py

```diff
@@ -242,7 +243,11 @@ def setup(
     if not session_name:
         raise RuntimeError('empty session name')
     # Resolve full tracing directory path
-    full_path = os.path.join(base_path, session_name)
+    # For init and runtime sessions, remove the suffixes in the full path
+    if is_init_session:
+        full_path = os.path.join(base_path, session_name.removesuffix('-init'))
```

Comment:
I think we discussed putting the snapshot session and the normal (runtime) session in the same directory (which would contain the 2 traces). Example:

* `~/.ros/tracing/${session_name}/${session_name}-snapshot/` for the snapshot (init events) trace
* `~/.ros/tracing/${session_name}/${session_name}/` for the normal (all events) trace

In this case, `base_path` would be `~/.ros/tracing/`. We would just need to do:

```suggestion
    # For init and runtime sessions, remove the suffixes in the full path
    if is_init_session:
        session_name_snapshot = session_name + '-snapshot'
        full_path = os.path.join(base_path, session_name, session_name_snapshot)
        session_name = session_name_snapshot
```

I think you _might_ have to create the `~/.ros/tracing/${session_name}/` directory, because I think LTTng only creates the trace directory (e.g., `~/.ros/tracing/${session_name}/${session_name}-snapshot/`) and expects the directory that contains it to exist.

Reply:
Changed the destination directory of snapshot session to `~/.ros/tracing/{session-name}/snapshot` (https://github.com/ros2/ros2_tracing/pull/191/commits/9ca2cfd84186c1fb375f95ae5b0530d52d6e5ac4)

Reply:
I am thinking modify `ros2 trace command` such that it will set the path of runtime traces to `~/.ros/tracing/{session-name}/runtime`, this will allow the users to open `~/.ros/tracing/{session-name}/` using babeltrace/tracecompass directly and also to distinguish between these two traces easily